src/frequenz/channels/_bidirectional.py

```diff
@@ -67,8 +75,26 @@ async def send(self, msg: V) -> None:
                     err.__cause__ = this_chan_error
 
         async def ready(self) -> None:
-            """Wait until the receiver is ready with a value."""
-            await self._receiver.ready()  # pylint: disable=protected-access
+            """Wait until the receiver is ready with a value.
+
+            Raises:
+                ReceiverStoppedError: if the receiver stopped producing messages.
+                ReceiverError: if there is some problem with the receiver.
+            """
+            try:
+                await self._receiver.ready()  # pylint: disable=protected-access
+            except ReceiverError as err:
+                # If this comes from a channel error, then we inject another
+                # ChannelError having the information about the Bidirectional
+                # channel to hide (at least partially) the underlaying
+                # Broadcast channels we use.
+                if isinstance(err.__cause__, ChannelError):
+                    this_chan_error = ChannelError(
+                        f"Error in the underlying channel {err.__cause__.channel}: {err.__cause__}",
+                        self._chan,  # pylint: disable=protected-access
+                    )
+                    this_chan_error.__cause__ = err.__cause__
+                    err.__cause__ = this_chan_error
 
```

Comment:
missing a raise here too?

Reply:
Yes, but that was fixed when moving the exceptions to `consume()` at least. I'll amend the original location to pretend I never make mistakes anyway :P